econml/validate/drtester.py

```diff
@@ -0,0 +1,796 @@
+    absolute difference in the mean CATE prediction and the overall ATE. These measures are then summed across bins,
+    weighted by a probability a unit is in each bin.
+
+    .. math::
+
+    Cal_G = \sum_k \pi(k) |E[s(Z) | k] - E[Y^{DR | k}|
+
+    Cal_O = \sum_k \pi(k) |E[s(Z) | k] - E[Y^{DR}|
+
+    The calibration r-squared is then defined as
+
+    .. math::
+
+    \mathcal{R^2}_C = 1 - \frac{Cal_G}{Cal_O}
+
+    The calibration r-squared metric is similar to the standard R-square score in that it can take any value
+    less than or equal to 1, with scores closer to 1 indicating a better calibrated CATE model.
+
+    **QINI**
+
+    Units are ordered by predicted CATE values and a running measure of the average treatment effect in each cohort is
+    kept as we progress through ranks. The QINI coefficient is then the area under the resulting curve, with a value
+    of 0 interpreted as corresponding to a model with randomly assigned CATE coefficients. All calculations are
+    performed on validation dataset results, using the training set as input.
+
+    More formally, the QINI curve is given by the following function:
+
+    .. math::
+
+    \tau_{QINI}(q) = Cov(Y^{DR}(g,p), 1\{\hat{\tau}(Z) \geq \hat{\mu}(q)\})
+
+    Where q is the desired quantile, \hat{\mu} is the quantile function, and \hat{\tau} is the predicted CATE function.
+    Y^{DR}(g,p) refers to the doubly robust outcome difference (relative to control) for the given observation.
+
+    The QINI coefficient is then given by:
+
+    .. math::
+
+    QINI = \int_0^1 \tau_{QINI}(q) dq
+
+    Parameters
+    ----------
+    reg_outcome: estimator
+        Nuisance model estimator used to fit the outcome to features. Must be able to implement `fit' and `predict'
+        methods
+
+    reg_t: estimator
+        Nuisance model estimator used to fit the treatment assignment to features. Must be able to implement `fit'
+        method and either `predict' (in the case of binary treatment) or `predict_proba' methods (in the case of
+        multiple categorical treatments).
+
+    n_splits: integer, default 5
+        Number of splits used to generate cross-validated predictions
+
+    References
+    ----------
+
+
+    [Chernozhukov2022] V. Chernozhukov et al.
+    Generic Machine Learning Inference on Heterogeneous Treatment Effects in Randomized Experiments
+    arXiv preprint arXiv:1712.04802, 2022.
+    `<https://arxiv.org/abs/1712.04802>`_
+
+    [Dwivedi2020] R. Dwivedi et al.
+    Stable Discovery of Interpretable Subgroups via Calibration in Causal Studies
+    arXiv preprint 	arXiv:2008.10109, 2020.
+    `<https://arxiv.org/abs/2008.10109>`_
+
+
+    [Radcliffe2007] N. Radcliffe
+    Using control groups to target on predicted lift: Building and assessing uplift model.
+    Direct Marketing Analytics Journal (2007), pages 14–21.
+    """
+
+    def __init__(
+        self,
+        reg_outcome,
```

Comment:
For consistency with the DRLearner API, let's call these `model_regression` and `model_propensity`. Just so that we don't introduce yet another naming convention in the package.